From: Paulo Alcantara <pc@cjr.nz>
Date: Sat, 17 Dec 2022 21:04:14 -0300
Subject: [PATCH] cifs: set correct status of tcon ipc when reconnecting
Git-commit: 25cf01b7c9200d6ace5a59125d8166435dd9dea7
References: bsc#1193629
Patch-mainline: v6.2-rc1

The status of tcon ipcs were not being set to TID_NEED_RECO when
marking sessions and tcons to be reconnected, therefore not sending
tree connect to those ipcs in cifs_tree_connect() and leaving them
disconnected.

Cc: stable@vger.kernel.org
Signed-off-by: Paulo Alcantara (SUSE) <pc@cjr.nz>
Signed-off-by: Steve French <stfrench@microsoft.com>
Acked-by: Paulo Alcantara <palcantara@suse.de>
---
 fs/cifs/connect.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs/cifs/connect.c b/fs/cifs/connect.c
index a66c7422b8bc..16da583536a4 100644
--- a/fs/cifs/connect.c
+++ b/fs/cifs/connect.c
@@ -262,8 +262,10 @@ cifs_mark_tcp_ses_conns_for_reconnect(struct TCP_Server_Info *server,
 			tcon->need_reconnect = true;
 			tcon->status = TID_NEED_RECON;
 		}
-		if (ses->tcon_ipc)
+		if (ses->tcon_ipc) {
 			ses->tcon_ipc->need_reconnect = true;
+			ses->tcon_ipc->status = TID_NEED_RECON;
+		}
 
 next_session:
 		spin_unlock(&ses->chan_lock);
-- 
2.39.0


